### ISSUE TITLE
With no scene specified on CLI, manim gives false choice.

### ISSUE THREAD

**Issue by cobordism** [ManimCommunity/manim#171]
I was running the command:
`manim -pl --media_dir ~/renderings ~/dev/manim/example_scenes/basic.py ` 
and was greeted by this:
```

[19:29:42] INFO     Read configuration files:                                                                      config.py:416
                    ['/home/cobordism/.local/lib/python3.8/site-packages/manim/default.cfg']                                    
1: GraphScene
2: LinearTransformationScene
3: MovingCameraScene
4: OpeningManimExample
5: ReconfigurableScene
6: SampleSpaceScene
7: SpecialThreeDScene
8: SquareToCircle
9: ThreeDScene
10: UpdatersExample
11: VectorScene
12: WarpSquare
13: WriteStuff
14: ZoomedScene

Choose number corresponding to desired scene/arguments.
(Use comma separated list for multiple entries)
Choice(s): 

```

Clearly something is wonky here The choices should have been only the scenes from basic.py.
i.e. OpeningManimExample, SquareToCircle, WarpSquare, WriteStuff, UpdatersExample.

**Comment by leotrs** [comment]
Interesting. I can't replicate this locally with the current master:

```
$ manim -pl basic.py
[09:49:01] INFO     Read configuration files: ['/home/leo/code/manim/manim/default.cfg']                                                                        config.py:416 
1: OpeningManimExample
2: SquareToCircle
3: UpdatersExample
4: WarpSquare
5: WriteStuff

Choose number corresponding to desired scene/arguments.
(Use comma separated list for multiple entries)
Choice(s):
```

**Comment by co-bordism** [comment]
try `./basic.py` instead of just `basic.py` 
or try from any other directory and using `some/path/to/basic.py`

**Comment by leotrs** [comment]
I'm at a loss here. Apparently when using `./basic.py`, manim is reading all Scene classes in the scope (which include all those imported with `from manim import *` e.g. GraphScene, ZoomedScene, etc); but when using `basic.py` these don't show up.

**Comment by Aathish04** [comment]
@leotrs I've found out why that happens.

---
 TL:DR:
`./basic.py` --> .replace(..) --> `..basic.py` --> regex --> ""

Also see:
https://regex101.com/r/SYBso1/2

---

> Apparently when using `./basic.py`, manim is reading all Scene classes in the scope (which include all those imported with `from manim import *` e.g. GraphScene, ZoomedScene, etc)
@leotrs 

The same thing happens when you move outside of the inner manim folder, and do 
`manim -pl manim/example_scenes/basic.py`

I believe it's because of the regex substitution here:

https://github.com/ManimCommunity/manim/blob/ea79842b17ccb7e0a5eb33be5edc5abb4ce60b50/manim/__main__.py#L137-L139

The pattern `"\..+$"` matches all characters after a `.` , and makes sure that the matched characters end with a newline, and replaces them with an empty string.

The pattern works fine when the only characters to the right of the `.` is `py`, but fails when you are leading in from another path.

For `./basic.py`, the `file_name.replace(os.sep,".")` makes the string into `..basic.py`, and the regex sees that the first character is a `.` and sees that the line ends with that string, and matches `..basic,py` entirely, and replaces it with an empty string.

`./basic.py` --> .replace(..) --> `..basic.py` --> regex --> ""

And then this empty string gets passed to `importlib`, which imports all the scenes that exist in the entire repo since it didn't receive a proper module to import.

If you are rendering the example scenes from outside the `example_scenes` directory, this is guaranteed to happen, since:

`manim/example_scenes.py.basic.py` --> .replace(..) --> `manim.example_scenes.py.basic.py` --> regex --> "manim"

In this case, it explicitly imports manim since that's all the regex left.

You can see this in action here: https://regex101.com/r/SYBso1/2

To fix the first case, where you're running the example_scenes inside the example_scenes folder, you can just add a `.lstrip(".")` after the `.replace(...)` since it'll remove those pesky `.`s.

For the second case, I think we might be better off using a completely different regex matcher altogether.
